timelock-corruption-detection/src/test/java/com/palantir/history/HistoryLoaderAndTransformerTest.java

```diff
@@ -0,0 +1,183 @@
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.history.remote.HistoryLoaderAndTransformer;
+import com.palantir.history.sqlite.SqlitePaxosStateLogHistory;
+import com.palantir.paxos.Client;
+import com.palantir.paxos.ImmutableNamespaceAndUseCase;
+import com.palantir.paxos.PaxosAcceptorState;
+import com.palantir.paxos.PaxosStateLog;
+import com.palantir.paxos.PaxosValue;
+import com.palantir.paxos.SqliteConnections;
+import com.palantir.paxos.SqlitePaxosStateLog;
+import com.palantir.timelock.history.HistoryQuery;
+import com.palantir.timelock.history.LogsForNamespaceAndUseCase;
+import com.palantir.timelock.history.PaxosLogWithAcceptedAndLearnedValues;
+
+public class HistoryLoaderAndTransformerTest {
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private static final Client CLIENT = Client.of("client");
+    private static final String USE_CASE = "useCase";
+    private static final String USE_CASE_LEARNER = "useCase!learner";
+    private static final String USE_CASE_ACCEPTOR = "useCase!acceptor";
+
+    private DataSource dataSource;
+    private PaxosStateLog<PaxosValue> learnerLog;
+    private PaxosStateLog<PaxosAcceptorState> acceptorLog;
+    private LocalHistoryLoader history;
+
+    @Before
+    public void setup() {
+        dataSource = SqliteConnections.getPooledDataSource(tempFolder.getRoot().toPath());
+        learnerLog = SqlitePaxosStateLog.create(
+                ImmutableNamespaceAndUseCase.of(CLIENT, USE_CASE_LEARNER), dataSource);
+        acceptorLog = SqlitePaxosStateLog.create(
+                ImmutableNamespaceAndUseCase.of(CLIENT, USE_CASE_ACCEPTOR), dataSource);
+        history = LocalHistoryLoader.create(SqlitePaxosStateLogHistory.create(dataSource));
+    }
+
+    @Test
+    public void canFetchLogsForQuery() {
+        writeToLogs(100);
+        int lastVerified = 27;
+        List<HistoryQuery> historyQueries = ImmutableList.of(HistoryQuery.of(
+                ImmutableNamespaceAndUseCase.of(CLIENT, USE_CASE), lastVerified));
+        List<LogsForNamespaceAndUseCase> paxosHistory
+                = HistoryLoaderAndTransformer.getLogsForHistoryQueries(history, historyQueries);
+
+        assertThat(paxosHistory.size()).isEqualTo(1);
+        LogsForNamespaceAndUseCase logsForNamespaceAndUseCase = paxosHistory.get(0);
+
+        assertThat(logsForNamespaceAndUseCase.getNamespaceAndUseCase().namespace()).isEqualTo(CLIENT);
+        assertThat(logsForNamespaceAndUseCase.getNamespaceAndUseCase().useCase()).isEqualTo(USE_CASE);
+        assertThat(logsForNamespaceAndUseCase.getLogs().size()).isEqualTo(100 - lastVerified);
+    }
+
+    @Test
+    public void canHandleDuplicateQueries() {
+        writeToLogs(100);
+        int minLastVerified = 27;
+
+        List<HistoryQuery> queries = IntStream.range(0, 10).boxed().map(
+                idx -> HistoryQuery.of(ImmutableNamespaceAndUseCase.of(CLIENT, USE_CASE), minLastVerified + idx))
```

Comment:
nit: This object's used quite a bit, you can just extract it as a constant and then use it. Also for the assertions you can then use `assertThat(bleh.getNamespaceAndUseCase()).isEqualTo(NAMESPACE_AND_USE_CASE)`